modules/ldap/src/main/java/com/neighborhoods/wasabi/ldap/impl/LdapDelegate.java

```diff
@@ -679,6 +680,110 @@ public DirectoryUser getDirectoryUserByUsername(CachedUserDirectory userDirector
         return null;
     }
 
+    /**
+     * Validate a directory token
+     * 
+     * @param userDirectory - the calling user directory
+     * @param username - The username to validate
+     * @param encodedPassword - The encrypted password to validate
+     * @return Always returns whether the token is valid or not
+     */
+    @Override
+    public boolean isDirectoryTokenValid(CachedUserDirectory userDirectory, String username, String encodedPassword) {
+        try {
+            DirectoryUser userInfo = userDirectory.lookupDirectoryUser(username);
+            // It is possible for the password to be null if it was retrieved outside of the login flow
+            if (userInfo != null && userInfo.getPassword() != null) {
+                // Encoded passwords are only in the cache
+                return userInfo.getPassword().equals(encodedPassword);
+            }
+        } catch (AuthenticationException ae) {
+            return false;
+        }
+        return false;
+    }
+
+    /**
+     * Populates the user cache with all Wasabi users
+     * 
+     * @param The calling CachedUserDirectory instance
+     * @throws AuthenticationException If unable to connect to ldap or bad LDAP queries
+     */
+    @Override
+    public void populateUserCache(CachedUserDirectory userDirectory) throws AuthenticationException {
+        try (LdapConnection connection = getLdapInfoConnection()) {
+            if (connection.isConnected() && connection.isAuthenticated()) {
```

Comment:
i've probably just forgotten, but what happens in the `else` case here?
or are these blocking until true and exception otherwise?